### PR TITLE
chore(dev): Add DEPRECATIONS.md file to track deprecations

### PR DESCRIPTION
**Pull request by jszwedko**
Add a file to track deprecations through their lifecycle and remind us to come back to take any
deprecation actions as part of each release.

See https://github.com/vectordotdev/vector/issues/14983 for the process proposal. The intent is to
make sure we follow through on deprecations, migrations, and removals.

Closes: https://github.com/vectordotdev/vector/issues/14983

Signed-off-by: Jesse Szwedko <jesse.szwedko@datadoghq.com>

### CODE REVIEW

File: docs/DEPRECATIONS.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# To be deprecated
+
+# To be migrated
+
+* legacy_openssl_provider v0.33.0 OpenSSL legacy provider flag should default to false
```
Comment (jszwedko) [inline docs/DEPRECATIONS.md:5]: Using this one as an example.

File: docs/DEPRECATIONS.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+# To be removed
+
+* legacy_openssl_provider v0.34.0 OpenSSL legacy provider flag should be removed
```
Comment (bruceg) [inline docs/DEPRECATIONS.md:9]: The process, as described, would have us only add this once the migration above is done. That is, a deprecation would only exist in one of the three blocks at any given time.

**Review by bruceg — COMMENTED** [review]
Do we want a simple script to ensure there aren't any deprecations listed for the current version to go along with this?

**Comment by jszwedko** [comment]
> Do we want a simple script to ensure there aren't any deprecations listed for the current version to go along with this?

We'd had that as "Future Work" on https://github.com/vectordotdev/vector/issues/14983 so I just added it as a checklist item for now.

File: docs/DEPRECATIONS.md (hunk 2 above)
Comment (jszwedko) [reply docs/DEPRECATIONS.md:9]: Ah, I see, that makes sense. Let me drop this one.

**Review by bruceg — APPROVED** [review]

**Pull request merged**